src/main/java/org/jabref/logic/texparser/DefaultTexParser.java

```diff
@@ -0,0 +1,181 @@
+package org.jabref.logic.texparser;
+
+import java.io.IOException;
+import java.io.LineNumberReader;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.jabref.model.database.BibDatabase;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.FieldName;
+import org.jabref.model.texparser.Citation;
+import org.jabref.model.texparser.TexParser;
+import org.jabref.model.texparser.TexParserResult;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DefaultTexParser implements TexParser {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DefaultTexParser.class);
+
+    /**
+     * It is allowed to add new cite commands for pattern matching.
+     */
+    private static final String[] CITE_COMMANDS = new String[] {"cite", "citep", "citet"};
+    private static final String CITE_REGEX = String.format("\\\\(?:%s)\\{([^\\}]*)\\}",
+            String.join("|", CITE_COMMANDS));
+
+    private static final String[] INCLUDE_COMMANDS = new String[] {"include", "includeonly", "input"};
+    private static final String INCLUDE_REGEX = String.format("\\\\(?:%s)\\{([^\\}]*)\\}",
+            String.join("|", INCLUDE_COMMANDS));
+
+    private final BibDatabase masterDatabase;
+
+    public DefaultTexParser(BibDatabase database) {
+        masterDatabase = database;
+    }
+
+    @Override
+    public TexParserResult parse(Path texFile) {
+        return parseTexFiles(new ArrayList<>(Collections.singletonList(texFile)));
```

Comment:
This is duplicated logic, either you use Arrays.asList or Collections.singletonList. Note that the singleton list is usually not modifiable.

Reply:
Oh, I did not read properly your comment before. Would it be OK if I use `new ArrayList<>(Arrays.asList(texFile))`?

`Arrays.asList` returns a fixed list, and `add` method is not allowed.

There are other choices. Or maybe I can do it in the traditional way:
```
List<Path> texFiles = new ArrayList<>();
texFiles.add(texFile);
return parseTexFiles(texFiles);
```

Reply:
`return parseTexFiles(Collections.singletonList(texFile));` should be sufficient (see below for the issue concerning `add`).